pkg/cfaws/granted_config.go

```diff
@@ -5,68 +5,54 @@ import (
 	"fmt"
 
 	"github.com/aws/aws-sdk-go-v2/config"
-	"github.com/bigkevmcd/go-configparser"
+	"gopkg.in/ini.v1"
 )
 
-type grantedSSOConfig struct {
-	granted_sso_start_url    string
-	granted_sso_region       string
-	granted_sso_account_name string
-	granted_sso_account_id   string
-	granted_sso_role_name    string
-	region                   string
-}
-
-func NewGrantedConfig(rawConfig configparser.Dict) *grantedSSOConfig {
-	return &grantedSSOConfig{
-		granted_sso_start_url:    rawConfig["granted_sso_start_url"],
-		granted_sso_region:       rawConfig["granted_sso_region"],
-		granted_sso_account_name: rawConfig["granted_sso_account_name"],
-		granted_sso_account_id:   rawConfig["granted_sso_account_id"],
-		granted_sso_role_name:    rawConfig["granted_sso_role_name"],
-		region:                   rawConfig["region"],
+func ParseGrantedSSOProfile(ctx context.Context, profile *Profile) (*config.SharedConfig, error) {
+	err := IsValidGrantedProfile(profile.RawConfig)
+	if err != nil {
+		return nil, err
 	}
-}
-
-func (gConfig *grantedSSOConfig) ConvertToAWSConfig(ctx context.Context, p *Profile) (*config.SharedConfig, error) {
-	cfg, err := config.LoadSharedConfigProfile(ctx, p.Name, func(lsco *config.LoadSharedConfigOptions) { lsco.ConfigFiles = []string{p.File} })
-	// if required profile doesn't exist then
-	// return empty config instead of error.
+	cfg, err := config.LoadSharedConfigProfile(ctx, profile.Name, func(lsco *config.LoadSharedConfigOptions) { lsco.ConfigFiles = []string{profile.File} })
 	if err != nil {
-		if _, ok := err.(config.SharedConfigProfileNotExistError); ok {
-			return &config.SharedConfig{}, nil
-		}
 		return nil, err
 	}
-
-	cfg.SSOAccountID = gConfig.granted_sso_account_id
-	cfg.SSORegion = gConfig.granted_sso_region
-	cfg.SSORoleName = gConfig.granted_sso_role_name
-	cfg.SSOStartURL = gConfig.granted_sso_start_url
-	cfg.Region = gConfig.region
```

Comment:
@JoshuaWilkes `cfg.Region` is an optional field but we should still populate the`cfg.Region` field if user has it available in "aws config" file. I remember some assume commands failing because `region` was not set.

Reply:
This still works as expected